pkg/cluster/cluster.go

```diff
@@ -110,31 +110,95 @@ func IsValidClusterKey(clusterKey string) bool {
 	return clusterKeyRE.MatchString(clusterKey)
 }
 
-func GetCluster(connection *sdk.Connection, clusterKey string) (*cmv1.Cluster, error) {
-	search := fmt.Sprintf("(display_name = '%s' "+
-		"or cluster_id = '%s' "+
-		"or external_cluster_id = '%s' )"+
-		"and status = 'Active'",
-		clusterKey, clusterKey, clusterKey)
-	response, err := connection.AccountsMgmt().V1().Subscriptions().List().Search(search).Size(1).Send()
+func GetCluster(connection *sdk.Connection, key string) (cluster *cmv1.Cluster, err error) {
+	// Prepare the resources that we will be using:
+	subsResource := connection.AccountsMgmt().V1().Subscriptions()
+	clustersResource := connection.ClustersMgmt().V1().Clusters()
+
+	// Try to find a matching subscription:
+	subsSearch := fmt.Sprintf(
+		"(display_name = '%s' or cluster_id = '%s' or external_cluster_id = '%s') and "+
+			"status in ('Reserved', 'Active')",
+		key, key, key,
+	)
+	subsListResponse, err := subsResource.List().
+		Search(subsSearch).
+		Size(1).
```

Comment:
If the key provided is the display name and multiple subscriptions match that display name, this will just return whichever is the first one provided in the API. That is probably fine behavior, but just something we should take note of.